third_party/move/move-model/src/builder/module_builder.rs

```diff
@@ -1319,16 +1321,34 @@ impl<'env, 'translator> ModuleBuilder<'env, 'translator> {
         is_pure
     }
 
-    fn deref_move_fun_types(&mut self, full_name: QualifiedSymbol, spec_fun_idx: usize) {
-        self.parent.spec_fun_table.entry(full_name).and_modify(|e| {
-            assert!(e.len() == 1);
-            e[0].params = e[0]
-                .params
-                .iter()
-                .map(|Parameter(n, ty)| Parameter(*n, ty.skip_reference().clone()))
-                .collect_vec();
-            e[0].result_type = e[0].result_type.skip_reference().clone();
-        });
+    fn deref_move_fun_types(
+        &mut self,
+        full_name: QualifiedSymbol,
+        spec_fun_idx: usize,
+        name_loc: &Loc,
+        name_value: move_symbol_pool::Symbol,
+    ) {
+        if let Some(entry) = self.parent.spec_fun_table.get_mut(&full_name) {
+            if entry.is_empty() {
+                self.parent.error(
+                    name_loc,
+                    &format!("missing declaration of `{}`", name_value),
+                );
+            } else if entry.len() > 1 {
```

Comment:
You could try to remove some code duplication by doing something like:
```                
                let error_msg = if entry.is_empty() {
                    "missing"
                } else {
                    "duplicate"
                };
                self.parent.error(
                    name_loc,
                    &format!("{} declaration of `{}` ", error_msg, name_value),
                );
```

Reply:
Done, thanks!